### ISSUE TITLE
Can I train with my own dataset?

### ISSUE THREAD

**Issue by Zarxrax** [uncbiag/SimpleClick#31]
How would I need to set up my dataset to be able to train with it? Do I need to modify some code?
Currently my dataset consists of images + alpha masks. What format would I need to convert it to?

**Comment by qinliuliuqin** [comment]
@Zarxrax Hi, yes, you can train on your own dataset. You need to create your own dataset. Please refer to https://github.com/uncbiag/SimpleClick/tree/v1.0/isegm/data/datasets

**Comment by Zarxrax** [comment]
Okay I think I see. I downloaded several of the datasets, and a couple of them look similar to mine, so I think it should be easy enough.

My next question is, when I run the training script, how do I specify which dataset to train on? I see in the config.yml I can put the path to my dataset. Would I simply delete any datasets from that file which I dont want to use, and then it will automatically choose mine?

**Comment by qinliuliuqin** [comment]
> Okay I think I see. I downloaded several of the datasets, and a couple of them look similar to mine, so I think it should be easy enough.
> 
> My next question is, when I run the training script, how do I specify which dataset to train on? I see in the config.yml I can put the path to my dataset. Would I simply delete any datasets from that file which I dont want to use, and then it will automatically choose mine?

No, you need to specify your training dataset. For example, if you use COCO+LVIS as the training set, you need to write it in the code, as shown here https://github.com/uncbiag/SimpleClick/blob/f33112e5a37ac97b9adfb9cda0106830e77b5b7d/models/iter_mask/plainvit_base448_cocolvis_itermask.py#L87

The config.yml only tells where to find the training data, for example, the COCO+LVIS dataset in the above case.

**Comment by Zarxrax** [comment]
Ah, perfect! Thank you.
